extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -94,6 +118,37 @@ class AmpYoutube extends AMP.BaseElement {
   }
 
   /** @private */
+  getPlayerReadyPromise_() {
+    return new Promise(resolve => {
+      this.getWin().addEventListener('message', event => {
+        const data = JSON.parse(event.data);
+        if (event.origin != 'https://www.youtube.com' ||
+            event.source != this.iframe_.contentWindow) {
+          return;
+        }
+        if (data.event == 'onReady') {
+          resolve(this.iframe_);
+        } else if (data.event == 'infoDelivery') {
```

Comment:
How about using a `deferred` value and then resolving that promise inside the youtube message handler? Would that work? Something like this:

```
layoutCallback() {
  ...
  this.playerReadyDeferred_ = Promise.defer();
  this.getWin().addEventListener('message', (event) => this.handleYouTubeMessages_(event));

  return loadPromise(iframe)
      .then(() => { return timer.promise(300); })
      .then(() => this.listenToFrame_())
      .then(() => this.deferredPlayerReady_.promise);
}
```

```
handleYouTubeMessages_() {
  ...
  if (data.event == 'onReady') {
    this.deferredPlayerReady_.resolve();
  }
  ...
}
```